api/graphite.go

```diff
@@ -819,11 +849,10 @@ func (s *Server) executePlan(ctx context.Context, orgId uint32, plan *expr.Plan)
 			if err != nil {
 				return nil, meta, err
 			}
-			series, err = s.clusterFindByTag(ctx, orgId, exprs, int64(r.From), maxSeriesPerReq-int(reqs.cnt), false)
+			findLimit := getClusterFindLimit(maxSeriesPerReq, int(reqs.cnt), len(rawReqs))
+			series, err = s.clusterFindByTag(ctx, orgId, exprs, int64(r.From), findLimit, false)
 		} else {
-			// find limit is the limit minus what we already consumed for other targets, adjusted for how many rawReqs we folded into this resolveSeriesRequest
-			// note that this doesn't account for duplicate requests like target=foo&target=foo because those are both represented by the same rawReq (see NewPlan)
-			findLimit := (maxSeriesPerReq - int(reqs.cnt)) / len(rawReqs)
+			findLimit := getClusterFindLimit(maxSeriesPerReq, int(reqs.cnt), len(rawReqs))
```

Comment:
Can this be factored out of the if/else block?

Reply:
yes. incoming...